Source/Shared/arcana/threading/task.h

```diff
@@ -271,6 +280,8 @@ namespace arcana
         using traits = internal::callable_traits<CallableT, void>;
         using wrapper = internal::input_output_wrapper<void, typename traits::error_propagation_type, false>;
 
+        auto cancel_pin = token.pin();
```

Comment:
Maybe just a simple comment saying that taking the pin early is safer? Something like that.